### PR TITLE
issue: 2670645 Add xlio extra api support

### PR DESCRIPTION
**Pull request by igor-ivanov**

### CODE REVIEW

File: src/vma-redirect.cpp [hunk 1]
```diff
@@ -79,12 +79,12 @@ daemon_fptr_t fn_daemon = NULL;
 sigaction_fptr_t fn_sigaction = NULL;
 
 ////////////////////////////////////////////////////////////////////////////////
-#define VMA_LOG_CB_ENV_VAR "VMA_LOG_CB_FUNC_PTR"
+#define XLIO_LOG_CB_ENV_VAR "VMA_LOG_CB_FUNC_PTR"
```
Comment (AlexanderGrissik) [inline src/vma-redirect.cpp:82]: Maybe change the  name itself to "XLIO_LOG_CB_FUNC_PTR"?

File: src/sockperf.cpp [hunk 2]
```diff
@@ -262,10 +262,10 @@ static const AOPT_DESC common_opt_desc[] = {
       aopt_set_string("no-rdtsc"),
       "Don't use register when taking time; instead use monotonic clock." },
     { OPT_LOAD_VMA,                                             AOPT_OPTARG,
-      aopt_set_literal(0),                                      aopt_set_string("load-vma"),
-      "Load VMA dynamically even when LD_PRELOAD was not used." },
+      aopt_set_literal(0),                                      aopt_set_string("load-xlio"),
+      "Load XLIO dynamically even when LD_PRELOAD was not used." },
     { OPT_RATE_LIMIT, AOPT_ARG, aopt_set_literal(0), aopt_set_string("rate-limit"),
-      "use rate limit (packet-pacing), with VMA must be run with VMA_RING_ALLOCATION_LOGIC_TX "
+      "use rate limit (packet-pacing), with XLIO must be run with VMA_RING_ALLOCATION_LOGIC_TX "
```
Comment (AlexanderGrissik) [inline src/sockperf.cpp:268]: VMA_RING_ALLOCATION_LOGIC_TX to XLIO_RING_ALLOCATION_LOGIC_TX ?

File: src/sockperf.cpp [hunk 3]
```diff
@@ -2073,7 +2073,7 @@ static char *display_opt(int id, char *buf, size_t buf_size) {
 }
 
 //#define ST_TEST
-// use it with command such as: VMA ./sockperf -s -f conf/conf.inp --vmarxfiltercb
+// use it with command such as: VMA ./sockperf -s -f conf/conf.inp --xliorxfiltercb
```
Comment (AlexanderGrissik) [inline src/sockperf.cpp:2081]: "... as: VMA ..." to "... as : XLIO ..." ?

File: src/sockperf.cpp [hunk 4]
```diff
@@ -3396,8 +3396,8 @@ packet pace limit = %d",
             s_user_params.fd_handler_type, s_user_params.mthread_server,
             s_user_params.sock_buff_size, s_user_params.threads_num, s_user_params.threads_affinity,
             s_user_params.is_blocked, s_user_params.is_nonblocked_send, s_user_params.do_warmup,
-            s_user_params.pre_warmup_wait, s_user_params.is_vmarxfiltercb,
-            s_user_params.is_vmazcopyread, s_user_params.mc_loop_disable, s_user_params.mc_ttl,
+            s_user_params.pre_warmup_wait, s_user_params.is_xliorxfiltercb,
+            s_user_params.is_xliozcopyread, s_user_params.mc_loop_disable, s_user_params.mc_ttl,
             s_user_params.uc_reuseaddr, s_user_params.tcp_nodelay,
             s_user_params.client_work_with_srv_num, s_user_params.b_server_reply_via_uc,
             s_user_params.b_server_dont_reply, s_user_params.b_server_detect_gaps,
```
Comment (AlexanderGrissik) [inline src/sockperf.cpp:3408]: See lines 3422, 3418
Also didn't find VMA_LIBRARY_MAJOR definition, in libxlio there something like PRJ_LIBRARY_MAJOR.

File: src/vma-redirect.cpp [hunk 5]
```diff
@@ -95,11 +95,11 @@ static vma_log_cb_t vma_log_get_cb_func() {
...
     sprintf(str, "%p", log_cb);
-    setenv(VMA_LOG_CB_ENV_VAR, str, 1);
+    setenv(XLIO_LOG_CB_ENV_VAR, str, 1);
 
     if (log_cb != vma_log_get_cb_func()) // verify that VMA will be able to read it correctly
     {
-        unsetenv(VMA_LOG_CB_ENV_VAR);
+        unsetenv(XLIO_LOG_CB_ENV_VAR);
         return false;
     }
```
Comment (AlexanderGrissik) [inline src/vma-redirect.cpp:105]: see line 176, not sure if it should stay like this ...

**Review by AlexanderGrissik — CHANGES_REQUESTED** [review]
Reviewed all the files and added some comments.
Also there is some modification needed in doc/main.dox which still describes old flags. like --load-vma.

**Comment by igor-ivanov** [comment]
Closed because final solution as https://github.com/Mellanox/sockperf/pull/198 is merged

**Pull request closed without merging**